hooks/tk-maya/render_media.py

```diff
@@ -69,42 +64,182 @@ def render(
                 name = os.path.basename(current_file_path)
 
         if not output_path:
-            output_path = self._get_temp_media_path(name, version, playblast_extension)
+            output_path = self._get_temp_media_path(name, version, "")
 
-        # The filename to use for the output of this playblast.
-        playblast_args["filename"] = output_path
-
-        if width:
-            # Width of the final image. This value will be clamped if larger than the width of the active window.
-            playblast_args["width"] = width
-
-        if height:
-            # Height of the final image. This value will be clamped if larger than the width of the active window.
-            playblast_args["height"] = height
+        playblast_args = self.get_default_playblastlast_args(output_path)
 
         self.logger.info(
             "Writing playblast to: %s using (%s)" % (output_path, playblast_args)
         )
 
-        maya.cmds.playblast(**playblast_args)
-        self.logger.info("Playblast written")
+        output_path = maya.cmds.playblast(**playblast_args)
+        self.logger.info("Playblast maybe written to %s" % output_path)
+
+        if os.path.exists(output_path):
+            self.logger.info("Playblast written to %s" % output_path)
+            return output_path
+
+        # Now, we did a playblast the the file is not on disk?
+        # What's happening is that if you render a movie, Maya append the movie name to the file name
+        # and if you render a file sequence, Maya append the sequence number and the extension to the file.
+        # Now we need to find the file on disk given the prefix we provided to the playblast command.
+
+        output_folder, output_file = os.path.split(playblast_args["filename"])
+
+        files = []
+        for f in os.listdir(output_folder):
+            try:
+                f_path = os.path.join(output_folder, f)
+                if not f.startswith(output_file) or not os.path.isfile(f_path):
+                    continue
+
+                # This method raise OSError if the file does not exist or is somehow inaccessible.
+                m_time = os.path.getmtime(f_path)
```

Comment:
It's good practice to limit the code of an `except` as much as possible to avoid hiding other errors that could be caught by a larger `try` section that shouldn't not be there in the first place.

Could we move the `try` much closer here, something like

```python
try:
    m_time = os.path.getmtime(f_path)
except OSError:
    pass
else:
     # Insert with a negative access time so the first element in the list is the most recent file
     heapq.heappush(files, (-m_time, f))
 ```